src/libsyntax/ast.rs

```diff
@@ -479,7 +451,7 @@ pub struct WhereEqPredicate {
 
 /// The set of MetaItems that define the compilation environment of the crate,
 /// used to drive conditional compilation
-pub type CrateConfig = Vec<P<MetaItem>>;
+pub type CrateConfig = HashSet<(Name, Option<Symbol>)>;
```

Comment:
I tried this actually (see https://github.com/rust-lang/rust/issues/28657), but wasn't able to observe stable improvements.